transports/ftp/src/test/java/org/mule/transport/ftp/FTPConnectorTestCase.java

```diff
@@ -110,7 +117,53 @@ public void testCustomFtpConnectionFactory() throws Exception
         assertEquals("Custom FTP connection factory has been ignored.", testObject, obj);
     }
 
+    @Test
+    public void isFile() throws Exception
+    {
+        FTPClient client = mock(FTPClient.class);
+        FTPFile file = mock(FTPFile.class);
+        FTPFile [] files = {file};
+        FTPListParseEngine engine = mock (FTPListParseEngine.class);
+        when(file.isFile()).thenReturn(true);
+        when(file.getName()).thenReturn("somePath");
+        when(client.initiateListParsing("somePath")).thenReturn(engine);
+        when(engine.getNext(2)).thenReturn(files);
+        FtpConnector connector = (FtpConnector)getConnector();
+
+        assertThat(connector.isFile("somePath", client), is(true));
+    }
+
+    @Test
+    public void isDirectory() throws Exception
+    {
+        FTPClient client = mock(FTPClient.class);
+        FTPFile file = mock(FTPFile.class);
+        FTPFile secondFile = mock(FTPFile.class);
+        FTPFile [] files = {file, secondFile};
+        FTPListParseEngine engine = mock (FTPListParseEngine.class);
+        when(file.isFile()).thenReturn(true);
+        when(client.initiateListParsing("somePath")).thenReturn(engine);
+        when(engine.getNext(2)).thenReturn(files);
+        FtpConnector connector = (FtpConnector)getConnector();
+
+        assertThat(connector.isFile("somePath", client), is(false));
+    }
+
+    @Test
+    public void isDirectoryWithOnlyAFile() throws Exception
+    {
+        FTPClient client = mock(FTPClient.class);
+        FTPFile file = mock(FTPFile.class);
+        FTPFile [] files = {file};
+        FTPListParseEngine engine = mock (FTPListParseEngine.class);
+        when(file.isFile()).thenReturn(true);
+        when(file.getName()).thenReturn("someFile");
+        when(client.initiateListParsing("somePath")).thenReturn(engine);
+        when(engine.getNext(2)).thenReturn(files);
```

Comment:
There is some duplicated code in the setup of the FTP objects. Extract to methods